docs/src/main/asciidoc/spanner.adoc

```diff
@@ -0,0 +1,149 @@
+----
+
+Configuring the `SpannerDialect` and a Cloud Spanner Driver class is typical of all Hibernate dialects in the `hibernate.properties` file:
+
+----
+hibernate.dialect=com.google.cloud.spanner.hibernate.SpannerDialect
+hibernate.connection.driver_class=com.google.cloud.spanner.jdbc.JdbcDriver
+hibernate.connection.url=jdbc:cloudspanner:/projects/{INSERT_PROJECT_ID}/instances/{INSERT_INSTANCE_ID}/databases/{INSERT_DATABASE_ID}
+----
+
+The service account JSON credentials file's location should be in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+
+The dialect and driver are compatible with all values (`create`, `create-drop`, and `update`) of the `hibernate.hbm2ddl.auto` setting.
+
+
+=== Spanner Dialect
+
+The `SpannerDialect` supports most of the standard Hibernate and Java Persistence annotations but there are important differences in features because of differences in https://cloud.google.com/spanner/docs/schema-and-data-model[Cloud Spanner's data model] from traditional SQL databases.
+
+
+==== Relationships
+
+The dialect supports all of the standard relational annotations:
+
+- `@OneToOne`
+- `@OneToMany`
+- `@ManyToOne`
+- `@ManyToMany`
+
+These can be used via `@JoinTable` or `@JoinColumn`.
+However, because Cloud Spanner does not support Foreign Keys, foreign-key-columns are just regular columns in Cloud Spanner.
+
+NOTE: The lack of foreign keys also means database-side cascading deletes are not supported via the `@OnDelete(action = OnDeleteAction.CASCADE)` annotation because there is no `ON DELETE CASCADE` constraint in Cloud Spanner DDL.
+However, Hibernate-side cascading operations such as `@ManyToOne(cascade = {CascadeType.ALL})` are supported.
+
+
+==== Constraints
+
+Cloud Spanner does not support constraints.
+As a result, `SpannerDialect` does not currently support any constraints such as `FOREIGN KEY`, `UNIQUE`, or `ON DELETE CASCADE`.
+
+NOTE: Hibernate doesn't directly rely on the existence of constraints to perform its operations and leaves the enforcement relationship links to the database.
+
+
+==== Collections
+
+`SpannerDialect` supports collection properties annotated with `@ElementCollection`.
+However, as with inter-object-relationships, foreign-key-columns in collection tables do not have the Foreign Key constraint.
+
+==== Table Catalogs and Schemas
+
+Cloud Spanner does not support table names with Catalog and Schema components:
+
+[source, java]
+----
+// Not supported: `public.store.book` is not a valid Cloud Spanner table name.
+@Table(
+  catalog = "public",
+  schema = "store",
+  name = "book"
+)
+
+// Supported.
+@Table(
+  name = "book"
+)
+----
+
+
+==== @ColumnDefault
+
+The dialect does not currently set default values based on the https://docs.jboss.org/hibernate/orm/5.4/javadocs/org/hibernate/annotations/ColumnDefault.html[`@ColumnDefault`] annotation,
+because `null` values aren't specially handled and are stored just like other values by Cloud Spanner and its driver.
```

Comment:
I guess another way to say it that Spanner does not support default column values, and hence the dialect doesn't support `@ColumnDefault`.